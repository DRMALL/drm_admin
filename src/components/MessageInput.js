
import React from 'react'
import Input from 'antd/lib/input'
import getInputValue from '../actions/getInputValue'
import Upload from 'antd/lib/upload'
import Modal from 'antd/lib/modal'
import Icon from 'antd/lib/icon'
import beforeUpload from '../utils/beforeUpload'
import LzEditor from 'react-lz-editor'


import { uploadImg, troubleImg } from '../commons/apis'
import dispatch from '../actions/dispatch'


export default props => {
  const { previewVisible, previewImage, fileList, title, abstract, content, isEdit, messageRichImg } = props.state.message
  const uploadButton = (
      <div>
        <Icon type="plus" style={{ fontSize: 28, marginBottom:12 }} />
        <div className="ant-upload-text" style={{ fontSize:12 }} >上传图片</div>
      </div>
  )
  const token = localStorage.getItem('token')
  const uploadProps = {
      action: `${troubleImg}?token=${token}`,
      name: 'bugs',
      onChange: onChange,
      listType: 'picture',
      fileList: messageRichImg,
      data: (file) => {
      },
      multiple: true,
      beforeUpload: beforeUpload,
      showUploadList: true
  }
  function onChange(e){
    let result = e.fileList
    result.map((item, index )=> {
      item.url = item.response ? item.response.data.url  : ''
      item.key = '-' + index
    })
    dispatch('MESSAGE_GET_RICH_TEXT_IMGAGES', result)
  }
  return (
    <div>
        <div className='message-new-tip' >标题</div>
        <Input onChange={ getInputValue } placeholder='输入信息标题'
               data-path='MESSAGE' data-id='title' value={ title } />
        <div className='message-new-tip' >内容</div>
        <LzEditor cbReceiver={ getMessageValue }  video={ false }
                audio={ false } urls={ false } importContent={ content }  active={isEdit}
                autoSave={ false }  uploadProps={uploadProps} />
        <div className='message-new-tip' >摘要（64字以内）</div>
        <Input.TextArea onChange={ getInputValue } rows={2} placeholder='输入摘要'
                        data-path='MESSAGE' data-id='abstract' value={ abstract } />
        <div className='message-new-tip' >封面图片（第一张为信息封面图）</div>
        <div className="clearfix" >
          <Upload
            action={`${uploadImg}?token=${localStorage.getItem('token')}`}
            name="news"
            listType="picture-card"
            fileList={fileList}
            multiple={ true }
            beforeUpload={ beforeUpload }
            onPreview={ handlePreview }
            onChange={ handleChange }
          >
            {fileList.length >= 10 ? null : uploadButton}
          </Upload>
          <Modal visible={ previewVisible } footer={null} onCancel={ handleCancel }>
            <img alt="example" style={{ width: '100%' }} src={ previewImage } />
          </Modal>
        </div>
    </div>
    )
}

function getMessageValue(e){
  dispatch('MESSAGE_GET_EDITOR_VALUE', { content: e })
}

function handlePreview(e) {
  console.log(e)
  let url
  url = e.response ? e.response.data.url : e.url
  dispatch('MESSAGE_PREVIEW_IMG', url)
}


function handleChange(e){
  dispatch('MESSAGE_UPLOAD_IMG', e.fileList)
}

function handleCancel(e){
  dispatch('MESSAGE_HIDDEN_MODAL')
}
