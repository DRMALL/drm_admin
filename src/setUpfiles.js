
//mock localStorage和react 的，只需要在文件中引入就行
import React from 'react';
if (typeof window !== 'undefined') {
    const matchMediaPolyfill = function matchMediaPolyfill() {
    return {
      matches: false,
      addListener() {
      },
      removeListener() {
      },
    };
  };
  window.matchMedia = window.matchMedia || matchMediaPolyfill;

  window.React = React;
  window.localStorage = ( function storageMock() {
    var storage = {};
    return {
      setItem: function(key, value) {
        storage[key] = value || '';
      },
      getItem: function(key) {
        return key in storage ? storage[key] : null;
      },
      removeItem: function(key) {
        delete storage[key];
      },
      get length() {
        return Object.keys(storage).length;
      },
      key: function(i) {
        var keys = Object.keys(storage);
        return keys[i] || null;
      }
    }
  })()
}
