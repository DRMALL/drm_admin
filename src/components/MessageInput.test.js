

import MessageInput from './MessageInput'
import React from 'react'
import renderer from 'react-test-renderer'
import setUpfiles from '../setUpfiles'


it('MessageInput test', () => {
  const state = { message: { previewVisible: false, fileList: [], title: '输入测试' } }
  const tree = renderer.create(<MessageInput state={ state } />)
  expect(tree).toMatchSnapshot()
})
