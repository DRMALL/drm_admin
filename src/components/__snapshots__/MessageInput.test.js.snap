// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput test 1`] = `
<div>
  <div
    className="message-new-tip"
  >
    标题
  </div>
  <input
    className="ant-input"
    data-id="title"
    data-path="MESSAGE"
    disabled={false}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="输入信息标题"
    type="text"
    value="输入测试"
  />
  <div
    className="message-new-tip"
  >
    内容
  </div>
  <textarea
    className="ant-input"
    data-id="content"
    data-path="MESSAGE"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="输入信息内容"
    rows={6}
    style={Object {}}
    value={undefined}
  />
  <div
    className="message-new-tip"
  >
    摘要（64字以内）
  </div>
  <textarea
    className="ant-input"
    data-id="abstract"
    data-path="MESSAGE"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="输入摘要"
    rows={2}
    style={Object {}}
    value={undefined}
  />
  <div
    className="message-new-tip"
  >
    封面图片（第一张为信息封面图）
  </div>
  <div
    className="clearfix"
  >
    <span
      className=""
    >
      <div
        className="ant-upload-list ant-upload-list-picture-card"
        style={undefined}
      />
      <div
        className="ant-upload ant-upload-select ant-upload-select-picture-card"
        style={
          Object {
            "display": "",
          }
        }
      >
        <span
          className="ant-upload"
          onClick={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={undefined}
          tabIndex="0"
        >
          <input
            accept=""
            multiple={true}
            onChange={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
          />
          <div>
            <i
              className="anticon anticon-plus"
            />
            <div
              className="ant-upload-text"
            >
              上传图片
            </div>
          </div>
        </span>
      </div>
    </span>
  </div>
</div>
`;
